internal/service/ec2/vpc.go

```diff
@@ -652,6 +665,50 @@ func resourceVPCUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("ipv6_cidr_block_network_border_group") {
```

Comment:
this logic should go nested in the change check for [assign_generated_ipv6_cidr_block](https://github.com/hashicorp/terraform-provider-aws/blob/d47e3f378862217487987d1cf7a2f22e2e1eff88/internal/service/ec2/vpc.go#L596)

Since `assign_gen..` is required to set network border group, its likely theyll both be set at the same time. currently you would assign a ipv6 to a block, then check again for border group then remove it and reassign it from a border group